planex/cmd/mock.py

```diff
@@ -50,6 +51,22 @@ def parse_args_or_exit(argv=None):
     return parser.parse_args(argv)
 
 
+def pty_check_call(cmd):
+    # python2.7 doesn't return the exitcode here:
+    pty.spawn(cmd)
+    # get exit status of first child
+    (pid, status) = os.waitpid(-1, 0)
+    if status != 0:
+        if os.WIFEXITED(status):
```

Comment:
I want to make sure I always raise an exception on non-zero exit status (both if program exits, gets killed, etc.). Would something like this look better?
```
+    returncode = 1
+    if status == 0:
+        returncode = 0
+    elif os.WIFEXITED(status):
+        returncode = os.WEXITSTATUS(status)
+        print "PID %d exited with status %d" % (pid, returncode)
+    elif os.WIFSIGNALED(status):
+        signal = os.WTERMSIG(status)
+        print "PID %d exited with signal %d" % (pid, signal)
+    else:
+        print "PID %d exited with non-zero status 0x%02x" % (pid, status)
+    if returncode > 0:
         raise subprocess.CalledProcessError(returncode, cmd)
```